content/03-reference/01-tools-and-interfaces/07-preview-features/001-mssql/001-mssql-start-from-scratch.mdx

```diff
@@ -0,0 +1,585 @@
+    ```
+
+  This creates a `package.json` with an initial setup for a Node.js app.
+
+1. Invoke the Prisma CLI by prefixing it with `npx` - confirm that you can see the Prisma help screen:
+
+  ```terminal
+  npx prisma
+  ```
+
+1. Set up your Prisma project by creating your [Prisma schema](../../reference/tools-and-interfaces/prisma-schema/prisma-schema-file) file with the following command:
+
+    ```terminal copy
+    npx prisma init
+    ```
+
+  This command created a new directory called `prisma` with the following contents:
+
+    - `schema.prisma`: The Prisma schema with your database connection and the Prisma Client generator
+    - `.env`: A [dotenv](https://github.com/motdotla/dotenv) file for defining environment variables (used for your database connection)
+
+</SwitchTech>
+
+## Connect your database
+
+To connect to your MSSQL database:
+
+1. Add the following line to the `client` block in your schema file to enable the MSSQL preview:
+
+    ```prisma file=prisma/schema.prisma
+    generator client {
+      provider        = "prisma-client-js"
+    |  previewFeatures = ["microsoftSqlServer"]
+    }
+    ```
+
+1. Set the `provider` and `url` fields of the `datasource` block in your Prisma schema as shown:
+
+  ```prisma file=prisma/schema.prisma
+  datasource db {
+    provider = "sqlserver"
+    url      = env("DATABASE_URL")
+  }
+  ```
+
+1. Define the `DATABASE_URL` [environment variable](../../reference/tools-and-interfaces/prisma-schema/prisma-schema-file#using-environment-variables) in the `prisma/.env` file - this is the connection string to your database.
+
+    The following example connection string [uses SQL authentication](mssql-local#enable-authentication-with-sql-logins-optional), but there are [other ways to format your connection string]((mssql-connection-string)):
+
+    ```js file=prisma/.env
+    DATABASE_URL="sqlserver://localhost:1433;database=mydb;user=sa;password=r@ndomP@$$w0rd;trustServerCertificate=true"
+    ```
+
+    > To get around TLS issues, add `encrypt=DANGER_PLAINTEXT` if you are connecting to MSSQL from MacOS specifically.
+
+1. Adjust the connection string to match your setup - see [MSSQL connection string](mssql-connection-string) for more information.
+
+1. Introspect your database to validate your connection string - the CLI will throw a `P4001` error because your database is empty: 
+
+  <CodeWithResult>
+
+  <Cmd>
+  
+  ```terminal
+  npx prisma introspect
+  ```
+
+  </Cmd>
+
+  <CmdResult>
+
+  ```
+  Environment variables loaded from prisma\.env
+  Prisma Schema loaded from prisma\schema.prisma
+
+  Introspecting based on datasource defined in prisma\schema.prisma …
+  Error: 
+  P4001 The introspected database was empty:
+  ```
+
+  </CmdResult>
+
+  </CodeWithResult>
+
+## Create database tables with SQL
+
+To create database tables, you can either:
+
+* Use the `sqlcmd` command line tool (Docker) OR
+* Use MSSQL Management Studio (Windows only)
```

Comment:
Might want to add [DataGrip](https://www.jetbrains.com/datagrip/) that is the best tool. :)

Reply:
Also, SQL Server Management Studio